src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -46,71 +49,128 @@ internal class WorkloadUpdateCommand : InstallingWorkloadCommand
-                _workloadManifestUpdater.UpdateAdvertisingManifestsAsync(_includePreviews, string.IsNullOrWhiteSpace(_fromCacheOption) ? null : new DirectoryPath(_fromCacheOption)).Wait();
-                Reporter.WriteLine();
-                Reporter.WriteLine(LocalizableStrings.WorkloadUpdateAdManifestsSucceeded);
-            }
-            else if (_printRollbackDefinitionOnly)
-            {
-                var workloadSet = WorkloadSet.FromManifests(_workloadResolver.GetInstalledManifests());
 
-                Reporter.WriteLine("==workloadRollbackDefinitionJsonOutputStart==");
-                Reporter.WriteLine(workloadSet.ToJson());
-                Reporter.WriteLine("==workloadRollbackDefinitionJsonOutputEnd==");
+                return null;
             }
-            else
+        }
+
+        public override int Execute()
+        {
+            WorkloadHistoryRecorder recorder = new WorkloadHistoryRecorder(_workloadResolver, _workloadInstaller);
+            recorder.HistoryRecord.CommandName = "update";
+
+            try
             {
-                try
+                if (!string.IsNullOrWhiteSpace(_downloadToCacheOption))
                 {
-                    UpdateWorkloads(_includePreviews, string.IsNullOrWhiteSpace(_fromCacheOption) ? null : new DirectoryPath(_fromCacheOption));
+                    try
+                    {
+                        var workloadIds = GetUpdatableWorkloads();
+                        recorder.HistoryRecord.WorkloadArguments = workloadIds.Select(id => id.ToString()).ToList();
+                        recorder.Run(() => DownloadToOfflineCacheAsync(new DirectoryPath(_downloadToCacheOption), _includePreviews, workloadIds).Wait());
+                    }
+                    catch (Exception e)
+                    {
+                        throw new GracefulException(string.Format(LocalizableStrings.WorkloadCacheDownloadFailed, e.Message), e, isUserError: false);
+                    }
                 }
-                catch (Exception e)
+                else if (_printDownloadLinkOnly)
                 {
-                    // Don't show entire stack trace
-                    throw new GracefulException(string.Format(LocalizableStrings.WorkloadUpdateFailed, e.Message), e, isUserError: false);
+                    var packageUrls = GetUpdatablePackageUrlsAsync(_includePreviews).GetAwaiter().GetResult();
+                    PrintDownloadLink(packageUrls);
+                }
+                else if (_adManifestOnlyOption)
+                {
+                    var workloadManifestInfo = _workloadResolver.GetInstalledManifests();
+                    recorder.HistoryRecord.WorkloadArguments = workloadManifestInfo.Select(m => m.Id).ToList();
+                    recorder.Run(() => _workloadManifestUpdater.UpdateAdvertisingManifestsAsync(_includePreviews, string.IsNullOrWhiteSpace(_fromCacheOption) ? null : new DirectoryPath(_fromCacheOption), workloadManifestInfo).Wait());
```

Comment:
I think this option also doesn't need a workload history record.  It does update the advertising manifests, but doesn't otherwise update or install workloads.  The advertising manifests are just there to let you know that there are newer workload versions available to update to.